lib/models/chat_history.dart

```diff
@@ -222,6 +224,70 @@ Stream<DeltaEvent> _handleDocumentChange(
           messageId: change.doc.id,
           pinned: false);
       yield AppendDeltaEvent(model);
+      break;
+    case "channel.hype_train.begin":
+      final model = TwitchHypeTrainEventModel(
+          pinned: true,
+          messageId: "train${data['event']['id']}",
```

Comment:
I'd move all these params `data['event']...` stuff to `TwitchHypeTrainEventModel` as getters. That way, you can do something like `messageId: model.id` and it'll stay consistent. We can probably do like `TwitchHypeTrainEventModel.fromFirestore(DocumentSnapshot<...>)` so this code looks like `final model = HypeTrainModel.fromDocumentChange(change.doc);`.

That would also allow us to write some tests 😉  Otherwise this looks good!

tbh we should probably do that for all of the models too. Would be nice to get tests here.

Reply:
Would the best way to test this be testing fractions of chat_history model?

Reply:
Nah, I'd write unit tests directly against the hype train model. Mostly verifying that the updating API works. We should one day simplify the chat_history model but it's not easy to test right now.

Reply:
Could it be better to not parse from `DocumentSnapshot` but from `Map<>`? So that the model doesnt have a dependency on the firebase package and that the tests dont require that sealed class?

Reply:
Yeah that would be good too :)